CUMULUS-2705: Add schemaspy generated doc/images (#2496)
nasa/cumulus: docs/architecture.md

Discovering data for ingest is handled via workflow step components using Cumulus `provider` and `collection` configurations and various triggers.    Data can be ingested from AWS S3, FTP, HTTPS and more.

#### Database

> Cumulus utililzes a user-provided PostgreSQL database backend with data replication to an elasticsearch instance utilized for API search query efficency.   For legacy reasons, Core is currently also deploying a DynamoDB datastore that writes are replicated to in parallel with the PostgreSQL database writes.   These data stores will be removed in future releases.

Author: Open to feedback re: this, it's intended to be a true, merge-able, but not replacement for @laurenfrederick 's RDS arch doc work.

Reviewer: ```suggestion
Cumulus utilizes a user-provided PostgreSQL database backend. For improved API search query efficiency Cumulus provides data replication to an Elasticsearch instance.   For legacy reasons, Cumulus is currently also deploying a DynamoDB datastore, and writes are replicated in parallel with the PostgreSQL database writes.   The DynamoDB replicated tables and parallel writes will be removed in future releases.
```

Reviewer: I had to read that first sentence a few times so reworded it. Then went on a rewording spree.
